Mathlib/CategoryTheory/Monoidal/Bimon_.lean

```diff
@@ -29,7 +29,46 @@ universe v₁ v₂ u₁ u₂ u
 
 open CategoryTheory MonoidalCategory
 
-variable (C : Type u₁) [Category.{v₁} C] [MonoidalCategory.{v₁} C] [BraidedCategory C]
+variable {C : Type u₁} [Category.{v₁} C] [MonoidalCategory.{v₁} C] [BraidedCategory C]
+
+open scoped Mon_Class Comon_Class
+
+/--
+A bimonoid object in a braided category `C` is a object that is simultaneously monoid and comonoid
+objects, and structure morphisms of them satisfy appropriate consistency conditions.
+-/
+class Bimon_Class (M : C) extends Mon_Class M, Comon_Class M where
+  mul_comul : μ[M] ≫ Δ[M] = (Δ[M] ⊗ Δ[M]) ≫ tensor_μ M M M M ≫ (μ[M] ⊗ μ[M]) := by aesop_cat
+  one_comul : η[M] ≫ Δ[M] = η[M ⊗ M] := by aesop_cat
+  mul_counit : μ[M] ≫ ε[M] = ε[M ⊗ M] := by aesop_cat
+  one_counit : η[M] ≫ ε[M] = 𝟙 (𝟙_ C) := by aesop_cat
+
+namespace Bimon_Class
+
+attribute [reassoc (attr := simp)] mul_comul one_comul mul_counit one_counit
+
+variable (M : C) [Bimon_Class M]
+
+/-- A version of `mul_comul`, where the arugument is explicit. -/
```

Comment:
There are similar typos below:
```suggestion
/-- A version of `mul_comul`, where the argument is explicit. -/
```
(I would suggested exchanging the names `mul_comul` and `mul_comul'`, etc, because it seems to me that that naming convention is more common in mathlib.)